concent_api/core/tests/test_core_views.py

```diff
@@ -496,6 +510,98 @@ def test_send_should_return_http_202_if_task_to_compute_deadline_is_correct(self
 
             self.assertIn(response_202.status_code, [200, 202])
 
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_task_to_compute_without_public_key_should_return_http_400(self):
+        assert StoredMessage.objects.count() == 0
+
+        compute_task_def = message.ComputeTaskDef()
+        compute_task_def['task_id']     = 'ABC00XYZ'
+        compute_task_def['subtask_id']  = 'ABC00XYZ'
+        compute_task_def['deadline']    = self.message_timestamp
+
+        task_to_compute                 = message.TaskToCompute(
+            compute_task_def     = compute_task_def,
+            provider_public_key  = PROVIDER_PUBLIC_KEY,
+            requestor_public_key = REQUESTOR_PUBLIC_KEY,
+        )
+
+        for field_name in [
+            'provider_public_key',
+            'requestor_public_key',
+        ]:
+            setattr(task_to_compute, field_name, None)
+
+            report_computed_task = message.tasks.ReportComputedTask(
+                task_to_compute = task_to_compute
+            )
+
+            correct_golem_data = message.ForceReportComputedTask(
+                report_computed_task = report_computed_task,
+            )
+
+            response = self.client.post(
+                reverse('core:send'),
+                data = dump(
+                    correct_golem_data,
+                    PROVIDER_PRIVATE_KEY,
+                    CONCENT_PUBLIC_KEY),
+                content_type = 'application/octet-stream',
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(PROVIDER_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_OTHER_PARTY_PUBLIC_KEY=b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii'),
+            )
+
+            self.assertEqual(response.status_code, 400)
+            self._assert_stored_message_counter_not_increased()
+            self._assert_client_count_is_equal(0)
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_task_to_compute_with_public_key_with_wrong_length_should_return_http_400(self):
```

Comment:
The same as above:
1) please use `self.task_to_compute` to get rid of biolerplate
2) please remove `for` loop and use `pytest` instead
3) inline unnecesary variable

Reply:
1,3. done
2. answered above.